checkov/common/bridgecrew/vulnerability_scanning/integrations/twistcli.py

```diff
@@ -71,19 +73,29 @@ async def report_results_async(
             {"Authorization": bc_api_key},
         )
 
-        async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(resolver=aiohttp.AsyncResolver())) as session:
-            async with session.post(
-                url=f'{bc_integration.api_url}{self.vulnerabilities_base_path}/results', headers=headers, json=payload
-            ) as response:
-                content = await response.text()
-
-            if response.ok:
-                logging.info(f"Successfully send report for package file {file_path}")
-                return 0
-            else:
-                logging.error(f"Failed to send report for package file {file_path}")
-                logging.error(f"Status code: {response.status}, Reason: {response.reason}, Content: {content}")
-                return 1
+        # adding retry mechanism for avoiding repeated unexpected Gateway Timeout from the server
+        logging.info(f"[twistcli](report_results_async) reporting results to the server")
+        request_max_tries = int(os.getenv('REQUEST_MAX_TRIES', 3))
+        sleep_between_request_tries = float(os.getenv('SLEEP_BETWEEN_REQUEST_TRIES', 1))
+
+        for i in range(request_max_tries):
+            logging.info(f"[twistcli](report_results_async) reporting attempt {i+1} out of {request_max_tries}")
+            async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(resolver=aiohttp.AsyncResolver())) as session:
+                async with session.post(
+                    url=f'{bc_integration.api_url}{self.vulnerabilities_base_path}/results', headers=headers, json=payload
+                ) as response:
+                    content = await response.text()
+
+                if response.ok:
+                    logging.info(f"Successfully send report for package file {file_path}")
+                    return 0
+                elif i != request_max_tries - 1:
+                    time.sleep(sleep_between_request_tries * (i + 1))
```

Comment:
please use the non blocking version https://docs.python.org/3/library/asyncio-task.html#sleeping